File: 심채윤/내려가기.js
const input1 = ['3', '1 2 3', '4 5 6', '4 9 0']
const input2 = ['3', '0 0 0', '0 0 0', '0 0 0']
const input = input2

const n = Number(input[0])
const arr = input.slice(1).map(el => el.split(' ').map(Number)).reverse()
let max = 0
let min = Infinity;

function solution(depth, result, indexs) {
  if (depth === n) {
    const sum = result.reduce((a, b) => a + b)
    max = Math.max(max, sum)
    min = Math.min(min, sum)
    return;
  }

  for (let i = 0; i < n; i++) {
    if (indexs[indexs.length - 1] === 0 && i === 2) continue
    if (indexs[indexs.length - 1] === 2 && i === 0) continue
    solution(depth + 1, [...result, arr[result.length][i]], [...indexs, i]);
  }
}

solution(0, [], [])
console.log(max, min);

File: 이연걸/내려가기.py
n = int(input())
board = []
for _ in range(n): board.append(list(map(int, input().split())))

max_d = [[0] * 3 for _ in range(n)]
for i in range(3): max_d[0][i] = board[0][i]

for i in range(1, n):
    max_d[i][0] = board[i][0] + max(max_d[i-1][0], max_d[i-1][1])
    max_d[i][1] = board[i][1] + max(max_d[i-1][0], max_d[i-1][1], max_d[i-1][2])
    max_d[i][2] = board[i][2] + max(max_d[i-1][1], max_d[i-1][2])
print(max(max_d[n-1]), end=' ')

for i in range(1, n):
    max_d[i][0] = board[i][0] + min(max_d[i-1][0], max_d[i-1][1])
    max_d[i][1] = board[i][1] + min(max_d[i-1][0], max_d[i-1][1], max_d[i-1][2])
    max_d[i][2] = board[i][2] + min(max_d[i-1][1], max_d[i-1][2])

print(min(max_d[n-1]), end='')
